core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -2090,6 +2093,12 @@ case class ReplicaLeaderElection(
   callback: ElectLeadersCallback = _ => {}
 ) extends ControllerEvent {
   override def state: ControllerState = ControllerState.ManualLeaderBalance
+
+  override def preempt(): Unit = callback(
+    partitionsFromAdminClientOpt.fold(Map.empty[TopicPartition, Either[ApiError, Int]]) { partitions =>
```

Comment:
nit: I think you can use `foreach` instead of fold + default value

Reply:
@mumrah i believe the initial `Map.empty[TopicPartition, Either[ApiError, Int]]` was used to structure the return type (`ElectLeadersCallback: Map[TopicPartition, Either[ApiError, Int]] => Unit`) which cannot be achieved using `foreach`.

Reply:
Yeah. The value returned by the `fold` is passed to `callback`. `foreach` would return `Unit`.